File: Programmers/src/Javascript/Lv0/특이한정렬.js
function solution(numlist, n) {
    return numlist.sort((a,b) => {
        let [gap1, gap2] = [Math.abs(a-n), Math.abs(b-n)];
        
        if(gap1===gap2)
            return b-a;
        else
            return gap1-gap2;
    })
}

File: Programmers/src/Java/Lv0/특이한 정렬.java
import java.util.*;

class Solution {
    public int[] solution(int[] numlist, int n) {
        Map<Integer, Integer> map = new HashMap<>();
        int[] answer = new int[numlist.length];
        
        Arrays.sort(numlist);
        
        for(int i=0; i<numlist.length; i++) {
            map.put(i, Math.abs(n - numlist[i]));
        }
        
        ArrayList<Integer> keyset = new ArrayList<>(map.keySet());
        keyset.sort((a,b) -> map.get(b).compareTo(map.get(a)));
        
        Collections.reverse(keyset);
        
        for(int i=0; i<answer.length; i++) {
            answer[i] = numlist[keyset.get(i)];
        }
        
        return answer;
    }
}